File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        int[] storageCharacters = new int[26];
        int left = 0;
        int maxCount = 0;
        int maxLength = 0;
        for (int right = 0; right < s.Length; right++) {
            int index = s[right] - 'A';
            storageCharacters[index]++;
            maxCount = Math.Max(maxCount, storageCharacters[index]);
            if ((right - left + 1) - maxCount > k) {
                storageCharacters[s[left] - 'A']--;
                left++;
            }
            maxLength = Math.Max(maxLength, right - left + 1);
        }
        return maxLength;
    }
}

File: Data Structures & Algorithms/reverse-string/submission-0.java
class Solution {
    public void reverseString(char[] s) {
        int right = 0, left = s.length - 1;
        while(right < left){
            char tmp = s[left];
            s[left] = s[right];
            s[right] = tmp;
            right++;
            left--;
        }
    }
}